interview-questions.md

```diff
@@ -9,59 +9,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
+  Your answer: `this` is a blank that is used to reference an object inside of itself. `this` also uses dot notation.
 
-  Researched answer:
+  Researched answer: `this` is a keyword that is only used on Objects. It can be used to target an objects key values. 
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a Javascript library that was created by Facebook to create a faster user interface. 
 
-  Researched answer:
+  Researched answer: React consists of a document object model and a virtual DOM. The virtual DOM a shadow for the DOM, listens for a diff between nodes, and reconciles only that particular node.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Render is the only required lifecycle method in a React class component. Data passed through the method render is displayed to the DOM
 
+  Researched answer: In a React class component, the Render method is a required lifecycle method. The Render method is always accompanied by a return. Multiple react data nested with fragments can be displayed to the DOM. 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a syntax that allows react to display Javascript data to the DOM by esentially merging HTML and Javascript together.
 
-  Researched answer:
+  Researched answer: JSX acts as an extension syntax that merges Javascript and HTML. Developers use JSX in React applications as it allows backend data to be accessed and displayed to the DOM.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager that contains data shared between developers such as jest. 
 
-  Researched answer:
+  Researched answer: Yarn allows developers to share and use code. Jest is a Javascript testing framework that can be accessed by developers by first running yarn add jest to their terminal. Yarn jest in the terminal allows the test to be performed.
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: An anonymous function is a function that does not have a delcared name.
 
-  Researched answer:
+  Researched answer: Anonymous functions can not be accessed after it is declared since the anonymous function does not have a name to reference.
 
 
 ## Looking Ahead: Terms for Next Week
 
-1. Conditional rendering:
+1. Conditional rendering: Sets parameters to be met before rending react elements
 
-2. Object-oriented programming:
+2. Object-oriented programming: A way of creating logic/behavior using only objects 
 
-3. Ruby:
+3. Ruby: Ruby is a programming language that uses only objects to perform logic.
 
-4. Ruby blocks:
+4. Ruby blocks: Anonymous functions used in methods.
 
-5. Ruby hashes:
+5. Ruby hashes: a collection of key values that acts similar to an array
```

Comment:
Your answers are solid. I encourage you to make them more conversational.  The more you can practice describing and explaining these terms and meanings, the better you will appear in an interview. Try adding in examples to show a deep level of understanding.  Great job!